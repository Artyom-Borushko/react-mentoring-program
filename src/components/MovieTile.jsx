import MovieCard from "./MovieCard";
import '../styles/movieTile.css';

export default function MovieTile({movies, clickHandler}) {

    return (
        <div className={'movie-tile'}>
            {movies.map((movie) => (
                <MovieCard
                    clickHandler={clickHandler}
                    movie={movie}
                    key={movie.id}
                />
            ))}
        </div>
    );
}
