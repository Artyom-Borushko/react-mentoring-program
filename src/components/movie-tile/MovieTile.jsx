import MovieCard from "./movie-card/MovieCard";
import './movieTile.css';

export default function MovieTile({movies, clickHandler}) {

    return (
        <div className={'movie-tile'}>
            {movies.map((movie) => (
                <MovieCard
                    clickHandler={clickHandler}
                    movie={movie}
                    key={movie.id}
                />
            ))}
        </div>
    );
}
