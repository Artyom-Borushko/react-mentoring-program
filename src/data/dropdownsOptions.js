export const movieDropdownOptions = ['edit', 'delete'];

export const moviesSortDropdownOptions = ['release date', 'title'];
