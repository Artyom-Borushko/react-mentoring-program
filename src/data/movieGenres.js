
export const listOfGenres = ['all', 'documentary', 'comedy', 'horror', 'crime'];
