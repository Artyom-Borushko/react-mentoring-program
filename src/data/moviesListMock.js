export const listOfMoviesMock = [
    { src: '/images/movies-list/pulpFiction.png', title: "Pulp Fiction", releaseYear: "2004", relevantGenres: ['action', 'crime'], rating: '9.2', duration: '2h 22min', description: 'Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!', id: "uniqueMovieId1" },
    { src: '/images/movies-list/bohemianRhapsody.png', title: "Bohemian Rhapsody", releaseYear: "2003", relevantGenres: ['documentary', 'drama'], rating: '8.4', duration: '1h 27min', description: 'Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!', id: "uniqueMovieId2" },
    { src: '/images/movies-list/killBillVol2.png', title: "Kill Bill: Vol 2", releaseYear: "1994", relevantGenres: ['comedy', 'crime', 'oscar'], rating: '6.2', duration: '3h 12min', description: 'Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!', id: "uniqueMovieId3" },
    { src: '/images/movies-list/avengersWarOfInfinity.png', title: "Avengers: War of Infinity", releaseYear: "2004", relevantGenres: ['action', 'adventure'], rating: '8.2', duration: '0h 56min', description: 'Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!', id: "uniqueMovieId4" },
    { src: '/images/movies-list/inception.png', title: "Inception", releaseYear: "2003", relevantGenres: ['action', 'adventure', 'horror'], rating: '10.0', duration: '1h 39min', description: 'Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!', id: "uniqueMovieId5" },
    { src: '/images/movies-list/reservoirDogs.png', title: "Reservoir dogs", releaseYear: "1994", relevantGenres: ['crime', 'oscar'], rating: '7.7', duration: '2h 00min', description: 'Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!', id: "uniqueMovieId6" },
];
