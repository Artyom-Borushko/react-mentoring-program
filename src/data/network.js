export default 'http://localhost:4000/movies';
