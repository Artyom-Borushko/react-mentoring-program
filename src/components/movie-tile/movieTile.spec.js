import {render, screen} from "@testing-library/react";
import React from "react";
import userEvent from "@testing-library/user-event";
import '@testing-library/jest-dom';
import MovieTile from "./MovieTile";
import {listOfMoviesMock} from "../../data/moviesListMock";

describe("MovieTile component tests", () => {

    it("renders MovieTile component", () => {
        const {container} = render(<MovieTile
            movies={listOfMoviesMock}
        />);

        expect(container).toMatchSnapshot();
    });

    it("event listener is working clicking on image", async () => {
        const mockClickHandler = jest.fn();
        render(<MovieTile
            movies={listOfMoviesMock}
            clickHandler={mockClickHandler}
        />);

        await userEvent.click(screen.getAllByRole('img', {name: 'movie card'})[0]);

        expect(mockClickHandler).toHaveBeenCalledTimes(1);
    });

    it("event listener is not working clicking on movie releaseYear", async () => {
        const mockClickHandler = jest.fn();
        render(<MovieTile
            movies={listOfMoviesMock}
            clickHandler={mockClickHandler}
        />);

        await userEvent.click(screen.getAllByText('2018')[0]);

        expect(mockClickHandler).toHaveBeenCalledTimes(0);
    });
});
