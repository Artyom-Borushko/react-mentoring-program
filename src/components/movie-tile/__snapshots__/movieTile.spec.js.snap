// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTile component tests renders MovieTile component 1`] = `
<div>
  <div
    class="movie-tile"
  >
    <div
      class="movie-card"
    >
      <div>
        <img
          alt="expand options"
          aria-label="expand options"
          class="movie-expand-options-icon movie options"
          src="images/options-selector/threeDotsExpandOptionsIcon.png"
        />
      </div>
      <img
        alt="movie card"
        aria-label="movie card"
        src="/images/movies-list/bohemianRhapsody.png"
      />
      <div
        class="movie-name-year-wrapper"
      >
        <p>
          Fifty Shades Freed
        </p>
        <p
          class="movie-release-year"
        >
          2018
        </p>
      </div>
      <p
        class="movie-genres"
      >
        Drama, Romance
      </p>
    </div>
    <div
      class="movie-card"
    >
      <div>
        <img
          alt="expand options"
          aria-label="expand options"
          class="movie-expand-options-icon movie options"
          src="images/options-selector/threeDotsExpandOptionsIcon.png"
        />
      </div>
      <img
        alt="movie card"
        aria-label="movie card"
        src="/images/movies-list/bohemianRhapsody.png"
      />
      <div
        class="movie-name-year-wrapper"
      >
        <p>
          Zootopia
        </p>
        <p
          class="movie-release-year"
        >
          2016
        </p>
      </div>
      <p
        class="movie-genres"
      >
        Animation, Adventure, Family, Comedy
      </p>
    </div>
    <div
      class="movie-card"
    >
      <div>
        <img
          alt="expand options"
          aria-label="expand options"
          class="movie-expand-options-icon movie options"
          src="images/options-selector/threeDotsExpandOptionsIcon.png"
        />
      </div>
      <img
        alt="movie card"
        aria-label="movie card"
        src="/images/movies-list/bohemianRhapsody.png"
      />
      <div
        class="movie-name-year-wrapper"
      >
        <p>
          Star Wars: The Last Jedi
        </p>
        <p
          class="movie-release-year"
        >
          2017
        </p>
      </div>
      <p
        class="movie-genres"
      >
        Fantasy, Adventure, Science Fiction
      </p>
    </div>
    <div
      class="movie-card"
    >
      <div>
        <img
          alt="expand options"
          aria-label="expand options"
          class="movie-expand-options-icon movie options"
          src="images/options-selector/threeDotsExpandOptionsIcon.png"
        />
      </div>
      <img
        alt="movie card"
        aria-label="movie card"
        src="/images/movies-list/bohemianRhapsody.png"
      />
      <div
        class="movie-name-year-wrapper"
      >
        <p>
          Black Panther
        </p>
        <p
          class="movie-release-year"
        >
          2018
        </p>
      </div>
      <p
        class="movie-genres"
      >
        Action, Adventure, Fantasy, Science Fiction
      </p>
    </div>
    <div
      class="movie-card"
    >
      <div>
        <img
          alt="expand options"
          aria-label="expand options"
          class="movie-expand-options-icon movie options"
          src="images/options-selector/threeDotsExpandOptionsIcon.png"
        />
      </div>
      <img
        alt="movie card"
        aria-label="movie card"
        src="/images/movies-list/bohemianRhapsody.png"
      />
      <div
        class="movie-name-year-wrapper"
      >
        <p>
          Coco
        </p>
        <p
          class="movie-release-year"
        >
          2017
        </p>
      </div>
      <p
        class="movie-genres"
      >
        Adventure, Comedy, Family, Animation
      </p>
    </div>
    <div
      class="movie-card"
    >
      <div>
        <img
          alt="expand options"
          aria-label="expand options"
          class="movie-expand-options-icon movie options"
          src="images/options-selector/threeDotsExpandOptionsIcon.png"
        />
      </div>
      <img
        alt="movie card"
        aria-label="movie card"
        src="/images/movies-list/bohemianRhapsody.png"
      />
      <div
        class="movie-name-year-wrapper"
      >
        <p>
          Ready Player One
        </p>
        <p
          class="movie-release-year"
        >
          2018
        </p>
      </div>
      <p
        class="movie-genres"
      >
        Adventure, Science Fiction, Action
      </p>
    </div>
    <div
      class="movie-card"
    >
      <div>
        <img
          alt="expand options"
          aria-label="expand options"
          class="movie-expand-options-icon movie options"
          src="images/options-selector/threeDotsExpandOptionsIcon.png"
        />
      </div>
      <img
        alt="movie card"
        aria-label="movie card"
        src="/images/movies-list/bohemianRhapsody.png"
      />
      <div
        class="movie-name-year-wrapper"
      >
        <p>
          Tomb Raider
        </p>
        <p
          class="movie-release-year"
        >
          2018
        </p>
      </div>
      <p
        class="movie-genres"
      >
        Action, Adventure
      </p>
    </div>
    <div
      class="movie-card"
    >
      <div>
        <img
          alt="expand options"
          aria-label="expand options"
          class="movie-expand-options-icon movie options"
          src="images/options-selector/threeDotsExpandOptionsIcon.png"
        />
      </div>
      <img
        alt="movie card"
        aria-label="movie card"
        src="/images/movies-list/bohemianRhapsody.png"
      />
      <div
        class="movie-name-year-wrapper"
      >
        <p>
          Thor: Ragnarok
        </p>
        <p
          class="movie-release-year"
        >
          2017
        </p>
      </div>
      <p
        class="movie-genres"
      >
        Action, Adventure, Fantasy
      </p>
    </div>
    <div
      class="movie-card"
    >
      <div>
        <img
          alt="expand options"
          aria-label="expand options"
          class="movie-expand-options-icon movie options"
          src="images/options-selector/threeDotsExpandOptionsIcon.png"
        />
      </div>
      <img
        alt="movie card"
        aria-label="movie card"
        src="/images/movies-list/bohemianRhapsody.png"
      />
      <div
        class="movie-name-year-wrapper"
      >
        <p>
          Beauty and the Beast
        </p>
        <p
          class="movie-release-year"
        >
          2017
        </p>
      </div>
      <p
        class="movie-genres"
      >
        Family, Fantasy, Romance
      </p>
    </div>
    <div
      class="movie-card"
    >
      <div>
        <img
          alt="expand options"
          aria-label="expand options"
          class="movie-expand-options-icon movie options"
          src="images/options-selector/threeDotsExpandOptionsIcon.png"
        />
      </div>
      <img
        alt="movie card"
        aria-label="movie card"
        src="/images/movies-list/bohemianRhapsody.png"
      />
      <div
        class="movie-name-year-wrapper"
      >
        <p>
          The Shape of Water
        </p>
        <p
          class="movie-release-year"
        >
          2017
        </p>
      </div>
      <p
        class="movie-genres"
      >
        Drama, Fantasy, Romance
      </p>
    </div>
  </div>
</div>
`;
